src/Peloton/ApiClient.cs

```diff
@@ -69,20 +69,20 @@ public async Task InitAuthAsync(string overrideUserAgent = null)
 
 				UserId = response.user_id;
 				SessionId = response.session_id;
-			} catch(Exception e)

```

Comment:
Maybe catch more specific error type to distinguish between an an authentication error and something else.